zboss/license.txt

```diff
@@ -38,3 +38,10 @@
  * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+ 
+/*
+ *  The following tags tell the "west ncs-sbom" tool which files are covered
+ *  by the license information above.
+ *  SPDX-License-Identifier:  LicenseRef-ZBOSS-Nordic-4-Clause
+ *  NCS-SBOM-Apply-To-File: ./**/*.a
```

Comment:
For now, the tool only provides information about a license and nothing about the copyright owners.

The license here is not a standardized license, so I had to create an ID for it somehow, so I followed the smimilar pattern as for Nordic 5-Caluse.

If you think that using `Nordic` in the ID may be confusing, we can think of something else, for example:
* skipping `Nordic`: `LicenseRef-ZBOSS-4-Clause`
* adding tool prefix will suggest that this ID is used only by our tool: `LicenseRef-west-ncs-sbom-ZBOSS-4-Clause`

What do you think about those IDs? Do you have different suggestion? They must start with the `LicenseRef-`.

Reply:
Since the license mixes DSR (copyright owner) and Nordic (the only hardware, on which the binary may be used), I think that the `LicenseRef-ZBOSS-4-Clause` is fine.

What I was also referring to, is that the output page (https://doki-nordic.github.io/test-demo-docs/temp/nrfxlib-binary.html#text-LICENSEREF-ZBOSS-NORDIC-4-CLAUSE) does not contain information about the copyright owner (starts with the license terms), which is crucial in case of ZBOSS (it is incorrect to assume that Nordic is the owner).

On the other hand - it might be worth asking if the license should be extended by the 5th clause. It was prepared for the nRF5 SDK, where there was no ZBOSS sources committed. Currently we distribute ZCL and BDB layers as sources, so there might be a need for the additional clause. Either way - this discussion is outside of the scope of this PR.

Reply:
I changed the license identifier, so it has no `Nordic` in it.

The `west ncs-sbom` tool does not provide copyright owner information in the report. For now, it only shows what kind of license is applied to the files. Since I removed the `Nordic` therm from the identifier, then noting suggests that the Nordic is the copyright owner.

At some point, we will probably implement also `copyright` reporting, so I will make sure that we are correctly reporting DSR as the copyright owner.

@tomchy, can you accept such solution?

Updated report is here: https://doki-nordic.github.io/test-demo-docs/temp/nrfxlib-binary.html

Reply:
That's fine from my perspective. Thank you for the update!